### PR TITLE
Add a way to explicitly mark a tuple list as a dict

### PR DESCRIPTION
**Pull request by jonludlam**
Signed-off-by: Jon Ludlam <jonathan.ludlam@citrix.com>

### CODE REVIEW

File: ppx/ppx_deriving_rpc.cppo.ml [hunk 1]
```diff
@@ -73,6 +73,8 @@ let attr_name  = attr_string "name"
 (* Documentation for variants / record members *)
 let attr_doc = attr_string "doc"
 
+let is_dict attr = match Ppx_deriving.attr ~deriver "dict" attr with Some _ -> [%expr true] | None -> [%expr false]
```
Comment (mseri) [inline ppx/ppx_deriving_rpc.cppo.ml]: You want to make my life harder when I drop the `ppx_deriving` dependecy 😆

File: ppx_test/test.ml [hunk 2]
```diff
@@ -229,9 +229,10 @@ let test_default_enum () =
   check_unmarshal_error enum_of_rpc (Rpc.Int 1L);
   check_unmarshal_error enum_of_rpc (Rpc.Float 1.0)
 
-type enum_string_map = (enum * string) list [@@deriving rpc]
+type enum2 = [`a | `b | `c] [@@deriving rpc]
+type enum_string_map = (enum2 * string) list [@dict] [@@deriving rpc]
```
Comment (mseri) [inline ppx_test/test.ml]: Please keep both tests: the one using `@dict` and the one not using it.

File: ppx/ppx_deriving_rpc.cppo.ml (hunk 1 above)
Comment (jonludlam) [reply ppx/ppx_deriving_rpc.cppo.ml]: Oh dear, really?

File: ppx_test/test.ml (hunk 2 above)
Comment (jonludlam) [reply ppx_test/test.ml]: Good point

**Comment by jonludlam** [comment]
I have force-pushed over the original commit - the review comment correctly pointed out that I had removed a test. This new commit makes it easier to see that I've now not removed it.

File: ppx/ppx_deriving_rpc.cppo.ml (hunk 1 above)
Comment (mseri) [reply ppx/ppx_deriving_rpc.cppo.ml]: No worries. I still hope we won’t need to drop it

**Pull request merged**